hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/RestCallCollector.java

```diff
@@ -0,0 +1,191 @@
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.ascii.rest;
+
+import com.hazelcast.internal.util.counters.MwCounter;
+
+import java.util.Comparator;
+import java.util.concurrent.ConcurrentSkipListSet;
+
+public class RestCallCollector {
+
+    public static final int SET_SIZE_LIMIT = 1000;
+
+    private static class RequestIdentifier implements Comparable<RequestIdentifier> {
+
+        public static final Comparator<RequestIdentifier> COMPARATOR = Comparator.comparing(RequestIdentifier::getMethod)
+                .thenComparing(RequestIdentifier::getPath);
+
+        private final String method;
+        private final String path;
+
+        RequestIdentifier(String method, String path) {
+            this.method = method;
+            this.path = path;
+        }
+
+        @Override
+        public int compareTo(RestCallCollector.RequestIdentifier o) {
+            return COMPARATOR.compare(this, o);
+        }
+
+        public String getMethod() {
+            return method;
+        }
+
+        public String getPath() {
+            return path;
+        }
+    }
+
+    private final MwCounter mapPostSuccCount = MwCounter.newMwCounter();
+    private final MwCounter mapPostFailCount = MwCounter.newMwCounter();
+    private final MwCounter mapGetSuccCount = MwCounter.newMwCounter();
+    private final MwCounter mapGetFailCount = MwCounter.newMwCounter();
+    private final MwCounter mapDeleteSuccCount = MwCounter.newMwCounter();
+    private final MwCounter mapDeleteFailCount = MwCounter.newMwCounter();
+    private final MwCounter mapTotalRequestCount = MwCounter.newMwCounter();
+    private final MwCounter queuePostSuccCount = MwCounter.newMwCounter();
+    private final MwCounter queuePostFailCount = MwCounter.newMwCounter();
+    private final MwCounter queueGetSuccCount = MwCounter.newMwCounter();
+    private final MwCounter queueGetFailCount = MwCounter.newMwCounter();
+    private final MwCounter queueDeleteSuccCount = MwCounter.newMwCounter();
+    private final MwCounter queueDeleteFailCount = MwCounter.newMwCounter();
+    private final MwCounter queueTotalRequestCount = MwCounter.newMwCounter();
+
+    private final MwCounter requestCount = MwCounter.newMwCounter();
+    private final ConcurrentSkipListSet<RequestIdentifier> uniqueRequests = new ConcurrentSkipListSet<>();
+    private final ConcurrentSkipListSet<String> accessedMaps = new ConcurrentSkipListSet<>();
+    private final ConcurrentSkipListSet<String> accessedQueues = new ConcurrentSkipListSet<>();
+
+    void collectExecution(RestCallExecution execution) {
+        updateRequestCounters(execution);
+        boolean isMap = "map".equals(execution.getObjectType());
+        boolean isQueue = "queue".equals(execution.getObjectType());
+        String objectName = execution.getObjectName();
+        updateAccessedObjectSets(isMap, isQueue, objectName);
+        if (execution.getMethod().equalsIgnoreCase("post")) {
```

Comment:
enum looks more suitable for `Method` instead of `String` . The string equality checks will be just a switch case here instead. 
Same goes for getObjectType as well. It could be either map or queue.